data/trie/patriciaMerkleTrie.go

```diff
@@ -348,6 +356,31 @@ func (tr *patriciaMerkleTrie) ResetOldHashes() [][]byte {
 	return oldHashes
 }
 
+// GetDirtyHashes returns all the dirty hashes from the trie
+func (tr *patriciaMerkleTrie) GetDirtyHashes() (data.ModifiedHashes, error) {
+	if tr.root == nil {
+		return nil, nil
+	}
+
+	err := tr.root.setRootHash()
+	if err != nil {
+		return nil, err
+	}
+
+	dirtyHashes := make(data.ModifiedHashes)
+	err = tr.root.getDirtyHashes(dirtyHashes)
+	if err != nil {
+		return nil, err
+	}
+
+	return dirtyHashes, nil
+}
+
+// SetNewHashes adds the given hashes to tr.newHashes
+func (tr *patriciaMerkleTrie) SetNewHashes(newHashes data.ModifiedHashes) {
+	tr.newHashes = newHashes
```

Comment:
Please add mutex protection everywhere tr.newHashes is being used.

Reply:
Done.